mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java

```diff
@@ -290,10 +291,19 @@ private void onTaskExecutorGatewayRequest(TaskExecutorGatewayRequest request) {
         if (state == null) {
             sender().tell(new Exception(), self());
         } else {
-            if (state.isRegistered() && state.getGateway().isDone()) {
-                sender().tell(state.getGateway().join(), self());
-            } else {
-                sender().tell(new Status.Failure(new Exception("")), self());
+            try {
+                if (state.isRegistered()) {
+                    sender().tell(state.getGateway().join(), self());
+                } else {
+                    sender().tell(new Status.Failure(new Exception("")), self());
+                }
+            } catch (Exception e) {
+                try {
+                    // let's try one more by reconnecting with the gateway.
+                    sender().tell(state.reconnect().join(), self());
```

Comment:
shall we try to reconnect more than once?

Reply:
This is a synchronous code path. Any reconnections cannot happen here because it blocks the queue.